jest-gae/src/main/java/com/searchly/jestgae/JestGaeClient.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    protected <T extends JestResult> HTTPRequest prepareRequest(final Action<T> clientRequest) throws MalformedURLException {
+        String elasticSearchRestUrl = getRequestURL(getNextServer(), clientRequest.getURI());
+        URL url = new URL(elasticSearchRestUrl);
+        boolean setAuthorizationHeader = false;
+
+        String userInfo = url.getUserInfo();
+        if (userInfo != null) {
+            String urlWithoutCredentials = elasticSearchRestUrl.replace(userInfo + "@", "");
+            url = new URL(urlWithoutCredentials);
+            setAuthorizationHeader = true;
+        }
+
+        HTTPRequest request = constructHttpMethod(clientRequest.getRestMethodName(), url, clientRequest.getData(gson));
+        log.debug("Request method={} url={}", clientRequest.getRestMethodName(), elasticSearchRestUrl);
+
+        // add headers to request
+        for (Entry<String, Object> header : clientRequest.getHeaders().entrySet()) {
+            request.addHeader(new HTTPHeader(header.getKey(), header.getValue().toString()));
+        }
+
+        if (setAuthorizationHeader) {
+            request.addHeader(new HTTPHeader("Authorization", "Basic " + Base64.encodeBase64String(userInfo.getBytes())));
+        }
+
+        return request;
+    }
+
+    protected HTTPRequest constructHttpMethod(String methodName, URL url, String payload) {
+        HTTPRequest httpRequest = null;
+        if (methodName.equalsIgnoreCase("POST")) {
+            httpRequest = new HTTPRequest(url, HTTPMethod.POST, fetchOptions);
+            log.debug("POST method created based on client request");
+        } else if (methodName.equalsIgnoreCase("PUT")) {
+            httpRequest = new HTTPRequest(url, HTTPMethod.PUT, fetchOptions);
+            log.debug("PUT method created based on client request");
+        } else if (methodName.equalsIgnoreCase("DELETE")) {
+            httpRequest = new HTTPRequest(url, HTTPMethod.DELETE, fetchOptions);
+            log.debug("DELETE method created based on client request");
+        } else if (methodName.equalsIgnoreCase("GET")) {
+            httpRequest = new HTTPRequest(url, HTTPMethod.GET, fetchOptions);
+            log.debug("GET method created based on client request");
+        } else if (methodName.equalsIgnoreCase("HEAD")) {
+            httpRequest = new HTTPRequest(url, HTTPMethod.HEAD, fetchOptions);
+            log.debug("HEAD method created based on client request");
+        }
+
+        if (httpRequest != null && payload != null) {
+            httpRequest.setPayload(payload.getBytes());
+            httpRequest.setHeader(new HTTPHeader("Content-type", REQUEST_CONTENT_TYPE));
+        }
+
+        return httpRequest;
+    }
+
+    private <T extends JestResult> T deserializeResponse(HTTPResponse response, Action<T> clientRequest, HTTPMethod httpMethod) throws IOException {
+        // If head method returns no content, content is added according to response code
+        String responseBody = null;
```

Comment:
Default responses for `HEAD` requests are removed as obsolete, please remove them from gae package too.